### ISSUE TITLE
Implement TryAsRef<ForwardToLeader<..>> for RaftError

### ISSUE THREAD

**Issue by HaHa421** [datafuselabs/openraft#1087]
**Is your feature request related to a problem? Please describe.**
this eases client implementation

**Describe the solution you'd like**
merged and backported

**Describe alternatives you've considered**
None

**Additional context**
```rust
impl<NID, N , E> TryAsRef<ForwardToLeader<NID, N>> for RaftError<NID, E>
where
    NID: NodeId,
    N: Node,
    E: Debug + TryAsRef<ForwardToLeader<NID, N>>,
{
    fn try_as_ref(&self) -> Option<&ForwardToLeader<NID, N>> {
        self.forward_to_leader()
    }
}
```

**Comment by drmingdrmer** [comment]
Make sense. Would you open a PR for it? :)

**Comment by drmingdrmer** [comment]
- Fixed in #1088 

Good job!
